src/readpool/mod.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod config;
+
+use futures::Future;
+use futures_cpupool as cpupool;
+
+pub use self::config::Config;
+
+pub struct WorkerThreadContext {
```

Comment:
How should we manage the `WorkerThreadContext` if we use this readpool?

Reply:
Updated. Please check out the latest future pool which supports `on_tick`.